### DISCUSSION TITLE
"Treat as MS1 Data" Option

### DISCUSSION THREAD

**Discussion by FlepTumbleweed** [metgem/metgem#99] [Q&A] [answered]
Hi there,

I recently started using MetGem, and it is working really well for me. Thanks for all your work. I was looking into the code to learn more about what calculations really happens in the background, and most of it was pretty clear. 👍 

However, I wasn't able to find out what the **"Treat as MS1 Data"**- Option within the "Process File"-window is doing exactly. To my understanding, MS2-data would be necessary for the Cosine calculation? Could you please explain how this changes data processing, or alternatively tell me in which part of the code I can find my answer?

Many thanks and best regards,
Flep

**Comment by n-elie** [comment, marked as answer]
Hello,

The "Treat as MS1 Data" is for MS2 spectra where you can't know the mass of the precursor ion, like Electronic Impact spectra, or real MS1 data.

The modified cosine score used in MetGem usually compares the spectra by finding matches between the fragments of two spectra, but also between the neutral losses (precursor _m/z_ - fragments _m/z_). The function will use the combination that will maximize the score.
But with these "MS1" spectra, the cosine scores are calculated by comparing only the fragments masses. The trick here is to fill the list of precursor ions with zeros: https://github.com/metgem/metgem/blob/885e7ccd58822fb35423adc970fe94ea4f17a880/metgem_app/workers/read_data.py#L59-L68

The function that will effectively do the cosine score computation is on a separate repository ([libmetgem](https://github.com/metgem/libmetgem)): https://github.com/metgem/libmetgem/blob/6281ffceb7ff132da7533198dc5de56a185d72c8/libmetgem/cosine.py#L30-L47
```python
dm = spectrum1_mz - spectrum2_mz

if dm == 0.:
    for i in range(spectrum1_data.shape[0]):
        for j in range(spectrum2_data.shape[0]):
            diff = spectrum2_data[j, MZ] - spectrum1_data[i, MZ]
            if abs(diff) <= mz_tolerance:
                scores.append((spectrum1_data[i, INTENSITY] * spectrum2_data[j, INTENSITY], i, j, SpectraMatchState.fragment))
else:
    for i in range(spectrum1_data.shape[0]):
        for j in range(spectrum2_data.shape[0]):
            diff = spectrum2_data[j, MZ] - spectrum1_data[i, MZ]
            if abs(diff) <= mz_tolerance:
                scores.append((spectrum1_data[i, INTENSITY] * spectrum2_data[j, INTENSITY], i, j, SpectraMatchState.fragment))
            elif abs(diff + dm) <= mz_tolerance:
                scores.append((spectrum1_data[i, INTENSITY] * spectrum2_data[j, INTENSITY], i, j, SpectraMatchState.neutral_loss))
```

As spectrum1_mz and spectrum2_mz are all zero, the cosine score calculation does not uses neutral losses.

Regards,
Nicolas

**Reply by FlepTumbleweed** [reply]
Hello,

Thank you so much for your quick and comprehensive answer, this makes a lot of sense. So I could try to generate networks from GC-EI-MS data as well, but it's probably not an option one would ever use with LC-ESI-MS2-data.

Best regards,
Flep
